src/main/java/recipeapplication/database/RecipeFirebaseCrud.java

```diff
@@ -22,21 +26,73 @@ public RecipeFirebaseCrud(Firestore firestore,
 
 	@Override
 	public void create(Recipe data) {
-		
-		this.firestore.collection(collection).document(data.getId()).set(data);
-		
+		System.out.println("In recipe creation section");
+
+		try {
+			if (this.firestore.collection(collection).document(data.getId()).get().get().exists() == false)  // Check if document already exists, to not allow overwriting it with create.
+				this.firestore.collection(collection).document(data.getId()).set(data);
+			else
+				throw new Exception("Failed to create, Recipe with this ID already exists!");
+		} catch (InterruptedException e) {
+			e.printStackTrace();
+		} catch (ExecutionException e) {
+			e.printStackTrace();
+		} catch (Exception e) {
+			System.out.println(e.getMessage());
+		}
 	}
 
 	@Override
-	public void read(String document) {
-		// TODO Auto-generated method stub
-		
+	public Recipe read(String document) {
+		System.out.println("In recipe reading section");
+		try {
+			DocumentSnapshot docSnap = this.firestore.collection(collection).document(document).get().get();
+			if (docSnap.exists() == false)
+				throw new Exception("Failed to read, recipe doesn't exist!");
+			else {
+				Map<String, Object> data = this.firestore.collection(collection).document(document).get().get().getData();
+
+				Recipe recipe = new Recipe();
```

Comment:
Suggestion: export the creation of Recipe to a new method

Reply:
Changed to use the constructor instead

Recipe recipe = new Recipe(
						data.get("id").toString(),
						data.get("category").toString(),
						((List<String>) data.get("ingridiant")),
						data.get("preperation").toString(),
						data.get("recipeImageId") != null ? data.get("recipeImageId").toString() : null);